### PR TITLE
HomeWork-jv-work-with-file

### PR DESCRIPTION
**Pull request by ViktoriiaSvidovska**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,40 @@
...
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String CSV_SPLIT_BY = ",";
 
+    public void getStatistic(String fromFileName, String toFileName) {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/WorkWithFile.java:12]: It is necessary to break the logic from this method into 3 separate methods:
1. Read data from file
2. We process this data, i.e. create a report
3. Write the report to a file

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,40 @@
...
+        int supplyAmount = 0;
+        int buyAmount = 0;
+        String line;
+        try (BufferedReader br = new BufferedReader(new FileReader(fromFileName));
+                BufferedWriter bw = new BufferedWriter(new FileWriter(toFileName))) {
+            while ((line = br.readLine()) != null) {
+                String[] data = line.split(CSV_SPLIT_BY);
+                String operation = data[0].trim();
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/WorkWithFile.java:20]: move indexes to the class constant

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,40 @@
...
+                } else if (operation.equals("buy")) {
+                    buyAmount += amount;
+                }
+            }
+            int result = supplyAmount - buyAmount;
+            bw.write("supply," + supplyAmount);
+            bw.newLine();
+            bw.write("buy," + buyAmount);
+            bw.newLine();
+            bw.write("result," + result);
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/WorkWithFile.java:33]:
```suggestion
```

It is necessary to write not one line at a time, but a fully formed report to a file

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,7 +1,40 @@
...
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading or writing file", e);
+        }
     }
 }
+
+
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/WorkWithFile.java:40]:
```suggestion
```

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,7 +1,61 @@
...
+    private String readFile(String fileName) {
+        StringBuilder contentBuilder = new StringBuilder();
+        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                contentBuilder.append(line);
+                contentBuilder.append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading file", e);
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:27]: add filename to message

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,7 +1,61 @@
...
+        reportBuilder.append("buy,").append(buyAmount).append(System.lineSeparator());
+        reportBuilder.append("result,").append(result).append(System.lineSeparator());
+        return reportBuilder.toString();
+    }
 
+    private void writeFile(String fileName, String report) {
+        try (BufferedWriter bw = new BufferedWriter(new FileWriter(fileName))) {
+            bw.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing file", e);
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:58]: same

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 7]
```diff
@@ -1,7 +1,61 @@
...
+        }
+        int result = supplyAmount - buyAmount;
+        reportBuilder.append("supply,").append(supplyAmount).append(System.lineSeparator());
+        reportBuilder.append("buy,").append(buyAmount).append(System.lineSeparator());
+        reportBuilder.append("result,").append(result).append(System.lineSeparator());
+        return reportBuilder.toString();
+    }
 
+    private void writeFile(String fileName, String report) {
+        try (BufferedWriter bw = new BufferedWriter(new FileWriter(fileName))) {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:55]:
```suggestion
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(fileName))) {
```

**Review by okuzan — APPROVED** [review]
Well done

**Pull request closed without merging**